entity-server-api/src/main/java/org/terracotta/entity/ActiveServerEntity.java

```diff
@@ -52,11 +52,16 @@
    *  called concurrently with other invokes.</p>
    *
    * @param clientDescriptor source instance from which the invocation originates.
+   * @param currentOrderedId current id this invoke is part of
+   * @param eldestOrderedId earliest active id for this client
    * @param message The message from the client
    * @return possible return value
    */
-  R invoke(ClientDescriptor clientDescriptor, M message) throws EntityUserException;
-  
+  R invoke(ClientDescriptor clientDescriptor,
+           long currentOrderedId,
```

Comment:
I wonder whether we could / should combine the extra parameters into a single object. This would reduce the noise in the method signature and also allow more flexibility for the future.

One wrinkle is that the passive wants that object to have the triple (clientDescriptor, currentOrderedId, eldestOrderedId), whereas the active already has the clientDescriptor.